lib/signaling/v2/icemediamonitor.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const { ICE_MONITOR_ACTIVITY_CHECK_PERIOD_MS, ICE_MONITOR_INACTIVITY_THRESHOLD_MS } = require('../../util/constants');
+
+/**
+ * @emits IceMediaMonitor#updated
+ */
+class IceMediaMonitor {
+  /**
+   * Construct a {@link IceMediaMonitor}.
+   */
+  constructor(pc, options) {
+    options = Object.assign({
+      activityCheckPeriodMS: ICE_MONITOR_ACTIVITY_CHECK_PERIOD_MS,
+      inactivityThresholdMS: ICE_MONITOR_INACTIVITY_THRESHOLD_MS,
+    }, options);
+
+    Object.defineProperties(this, {
+      _pc: {
+        value: pc
+      },
+      _timer: {
+        value: null,
+        writable: true,
+      },
+      _activityCheckPeriodMS: {
+        value: options.activityCheckPeriodMS
+      },
+      _inactivityThresholdMS: {
+        value: options.inactivityThresholdMS
+      }
+    });
+  }
+
+  // TODO: remove this before merging.
+  _makarandlog() {
+    var newArgs = ['makarand: '];
+    var args = Array.prototype.slice.call(arguments);
+    console.log.apply(console.log, newArgs.concat(args));
+  }
+
+  _getMediaStats() {
```

Comment:
Although it may not be necessary in this case - I thought using twilio-webrtc was preferred because it ensures standardized response across browsers. But I am open to using peerConnection directly.